src/re_graph/internals.cljc

```diff
@@ -140,17 +140,20 @@
                                                                "Accept" "application/json"})
                                        (merge {:body (encode payload)
                                                :as :json
+                                               :coerce :always
```

Comment:
The lack of this option is the root cause of the corruption. By default clj-http will **not** transform error responses into JSON. This causes a string representation to be passed to `insert-http-status`, which then always overwrites the server's response with the default.